dashboard/db/migrate/20170127183604_create_sign_ins.rb

```diff
@@ -0,0 +1,13 @@
+class CreateSignIns < ActiveRecord::Migration[5.0]
+  def change
+    create_table :sign_ins do |t|
+      # The user to which the sign-in applies.
+      t.belongs_to :user, null: false
+      # The timestamp of the sign-in. The column is indexed so as to be able to
+      # quickly compute 7-DA and 30-DA counts.
+      t.timestamp :sign_in_at, null: false, index: true
+      # The number of total sign-ins (including this one) for this user.
+      t.integer :sign_in_count, null: false
```

Comment:
Note that this data is redundant with `users.sign_in_count`, and could also be recalculated whenever necessary from the rows in this table. I'd suggest removing this column unless there is a specific use-case (query optimization) in mind where the redundancy will be needed (or if you have another specific plan in mind, such as migrating away from the corresponding column from the `users` table), due to the additional overhead of keeping this data in sync (and figuring out priority if/when the data gets out of sync in certain edge cases).